src/components/CarListComponent/BodyList/BodyList.tsx

```diff
@@ -0,0 +1,48 @@
+import { Query_GetCarsQuery } from '../../../api/graphql/__generated__/graphql-types';
+import ListImageItem from '../ListImageItem/ListImageItem';
+import ListInfoItem from '../ListInfoItem/ListInfoItem';
+import { Container, Section } from './BodyListStyled';
+import { defaultImagePath } from '../../../helpers/objectValues';
+interface BodyListProps {
+  carsInfoArray: Query_GetCarsQuery;

```

Comment:
Probably better to just pass in `cars`. Decouple your components from api implementation details as much as possible. If you were building this component first, before you know what the api result structure was going to be you would never think to have a prop like this. You'd just say that this component takes a cars array. You probably don't even want to couple it with an api type, just create a `Car` based on what this component needs and typescript will tell you if the cars you pass in are wrong or not

Reply:
Thank you for this, i cheked and i had a problem with a type generated by CodeGen, thats why i couldn't pass as Cars array because the response property and the generated type Cars did not match.